### DISCUSSION TITLE
RLS? User-based Row Filtering?

### DISCUSSION THREAD

**Discussion by horrorstrum** [Budibase/budibase#12415] [Help] [answered]
I am trying to build a simple gift exchange app for my family. When a user logs in, they are given the option to edit their own list or someone else's. If they edit their own list, they are only allowed to see the rows that they've created, and any rows created by other users are hidden. If they edit someone else's list, they are allowed to see their own rows and the rows of other users.

I am struggling with the implementation of the row filtering. 
In the Users table I have relational columns for both "Recipient" and "Posted By" that relate to the Gift_ID in the Gifts table, so the logic should be fairly straightforward. However so far I am only able to show/hide entire tables based on this logic, and even then it doesn't quite work the way I would like it to.

Should this be implemented with filters, conditions, or RLS? Does BudiBase even have RLS? This is my first time using this tool so I am learning as I go. Any help would be greatly appreciated.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @horrorstrum,

I have created a POC of what I believe you are asking. You can download this [here](https://github.com/Budibase/budibase/files/13449727/gift.app-export-1700741820666.tar.gz). In this app I've made use of the `internal Budibase db`, `Url variables`, `Formulas` for pulling out user column and Auto created by `_ids`. 

A few different components are used such as; `table block` and `form block` component. Other aspects that were used include `Navigate to` actions along with the url variables to pass the correct ids to the right pages. Additionally the `current user` binding was used within filters so that only certain data is shown on specific pages. I hope that this at least points you in the right direction with building your app. 

The breakdown of this app includes 3 different pages; 
1. Homepage from here you can see 2 different tables
2. 1 showing all users within the app, with buttons that perform different actions
3. 2 showing all records created by the user
4. A screen for viewing records related to whatever user `_id` is passed to it
5. A screen for creating records related to a specific user

Lastly to answer your last question no we don't have any functionality in regard RLS.

I hope this helps.
